### ISSUE TITLE
Support Roi

### ISSUE THREAD

**Issue by thorstenwagner** [locked-fg/JFeatureLib#36]
Hi,

thank you for this nice library which saved me a lot of time.

However, it would be nice if one could set ROIs for processing an image. At the moment I crop the image to the Roi and then apply the selected feature. However, it would be faster just to use the Roi during calculation.

Best,
Thorsten

**Comment by locked-fg** [comment]
Hi Thorsten,
have you had a look at https://github.com/locked-fg/JFeatureLib/wiki/HowTo#using-masks ?
Cheers Franz

**Comment by thorstenwagner** [comment]
No, but this option is for command line, right? I use JFeatureLib as java library inside my programm.

**Comment by locked-fg** [comment]
Right, it's a cmd line option but you can have a look at the code how it's done. Unfortunately it's so long ago that I wrote the code that I'd have to look into the code closely myself to tell you.

**Comment by locked-fg** [comment]
It's actually not that complicated: Have a look at
https://github.com/locked-fg/JFeatureLib/blob/master/src/main/java/de/lmu/ifi/dbs/jfeaturelib/utils/Extractor.java
And where /how the masking option is used.
